### PR TITLE
feat(components/forms): checkbox errors easy mode

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1923/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1923/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1923/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1923/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1923/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1923/playground/)

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.html [hunk 1]
```diff
@@ -51,5 +51,20 @@
...
+  </ng-container>
+  <ng-template #labelElement>
+    <ng-content select="sky-checkbox-label" />
+  </ng-template>
 </label>
+<div *ngIf="labelText && ngControl?.errors">
+  <sky-form-errors
+    [id]="errorId"
+    [errors]="ngControl?.errors"
+    [labelText]="'skyux_checkbox_required_label_text' | skyLibResources"
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/forms/src/lib/modules/checkbox/checkbox.component.html:65]: the `labelText` used to see if user is going to be using easy mode checkbox would have more context on the checkbox right? why not use that in the `sky-form-errors`'s `labelText` to give more context in the error?

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.html [hunk 2]
```diff
@@ -51,5 +51,20 @@
...
+    <sky-checkbox-label>{{ labelText }}</sky-checkbox-label>
+  </ng-container>
+  <ng-template #labelElement>
+    <ng-content select="sky-checkbox-label" />
+  </ng-template>
 </label>
+<div *ngIf="labelText && ngControl?.errors">
+  <sky-form-errors
+    [id]="errorId"
+    [errors]="ngControl?.errors"
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/forms/src/lib/modules/checkbox/checkbox.component.html:64]: wouldn't it be better to send this input for `sky-form-error` only the value for `this.required` to be more explicit about which error we're sending

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.html (hunk 1 above)
Comment (Blackbaud-ErikaMcVey) [reply libs/components/forms/src/lib/modules/checkbox/checkbox.component.html:65]: Adam and I decided against including that--it looks weird. With a typical form control the input provides some space between the error and the label. With the checkbox it is directly below, so it looks kind of overkill

image

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.html (hunk 2 above)
Comment (Blackbaud-ErikaMcVey) [reply libs/components/forms/src/lib/modules/checkbox/checkbox.component.html:64]: This section also projects any custom validation error messages the consumer provides via the ng-content. We're not sure what their errors are so we can't be more specific than that. I'm not entirely sure _what_ custom validation someone might need to do on a checkbox, but I'm sure there is something 😂

Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/forms/src/lib/modules/checkbox/checkbox.component.html:64]: right but `sky-form-error` would consume and display that from an `ng-content` inside of it and not from this input afaik

File: libs/components/forms/src/lib/modules/checkbox/checkbox.component.html (hunk 1 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/forms/src/lib/modules/checkbox/checkbox.component.html:65]: ok yup i see your point 😂

**Review by Blackbaud-SandhyaRajasabeson — APPROVED** [review]

**Pull request merged**